armi/reactor/blocks.py

```diff
@@ -1529,6 +1530,20 @@ def rotate(self, deg):
         """
         raise NotImplementedError
 
+    def setAxialExpTargetComp(self, targetComponent):
+        """sets the targetComponent for the axial expansion changer
+
+        Parameter
+        ---------
+        targetComponent: :py:class:`Component <armi.reactor.components.component.Component>` object
+            component specified to be target component for axial expansion changer
+
+        See Also
+        --------
+        armi.reactor.converters.axialExpansionChanger.py::ExpansionData::_setTargetComponents
+        """
+        self.axialExpTargetComponent = targetComponent
```

Comment:
See comment about the parameter above. I also recommend that we check that on assignment of the value that this against the available components. Additionally, this method can be embedded into the parameters and it can be set of b.p.whatever.

Also, I'd add this check as part of the parameter's `setter`.

Reply:
> I also recommend that we check that on assignment of the value that this against the available components

This occurs in `blockBlueprint.py`. If the component requested isn't available, a Runtime Error is thrown. This could be moved to a parameter's setter.